packages/block-editor/src/components/link-control/index.js

```diff
@@ -219,20 +255,52 @@ function LinkControl( {
 			const isInternal = startsWith( val, '#' );
 			const isTel = protocol.includes( 'tel' );
 
-			const handleManualEntry =
+			const maybeURL =
 				isInternal ||
 				isMailto ||
 				isTel ||
 				isURL( val ) ||
 				( val && val.includes( 'www.' ) );
 
-			return handleManualEntry
+			return maybeURL
 				? handleDirectEntry( val, args )
 				: handleEntitySearch( val, args );
 		},
 		[ handleDirectEntry, fetchSearchSuggestions ]
 	);
 
+	const handleOnCreate = async ( entityTitle ) => {
+		let newEntity;
+
+		setIsResolvingLink( true );
+		setErrorMsg( null );
+		try {
+			newEntity = await createEntity( 'page', entityTitle );
+		} catch ( error ) {
+			setErrorMsg(
+				error.msg ||
+					__(
+						'An unknown error occurred during Page creation. Please try again.'
+					)
+			);
+		}
+
+		setIsResolvingLink( false );
+
+		if ( newEntity ) {
+			// only set if request is resolved
+			onChange( newEntity );
+			setIsEditingLink( false );
+		} else {
+			setIsEditingLink( true );
+		}
+	};
+
+	const handleSelectSuggestion = ( suggestion, _value = {} ) => () => {
```

Comment:
Per previous comment:

>Why is `handleSelectSuggestion` assigned as a function inside a function, if we're only ever just calling it like this?

```suggestion
	const handleSelectSuggestion = ( suggestion, _value = {} ) => {
```

Reply:
Thank you for catching this.

I refactored the code here and therefore the previously necessary partial application became redundant but I failed to remove it.

You'll have to excuse this. It's largely due to the complexity of the rebases that I've encountered due to the rapidly changing nature of this component.